spec/rubocop/cop/rails/validations_grouping_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::ValidationsGrouping, :config do
+  let(:cop_config) { { 'SingleAttributeValidations' => false } }
+
+  it 'registers an offense when attribute validations are split' do
+    expect_offense(<<~RUBY)
+      validates :name, :email, presence: true
+                       ^^^^^^ Group together all email validations.
+      validates :email, format: { with: /@/ }
+                ^^^^^^ Group together all email validations.
+    RUBY
+  end
+
+  it 'does not register an offense when attribute validations are grouped' do
+    expect_no_offenses(<<~RUBY)
+      validates :email, presence: true, format: { with: /@/ }
+    RUBY
+  end
+
+  it 'does not register an offense for not-literal attribute name validations' do
+    expect_no_offenses(<<~RUBY)
+      validates attribute, presence: true
+      validates attribute, format: { with: /@/ }
+    RUBY
+  end
+
+  it 'does not register an offense when attribute validations have different configuration options' do
+    expect_no_offenses(<<~RUBY)
+      validates :email, presence: true, if: :some_condition?
+      validates :email, format: { with: /@/ }
+    RUBY
+  end
+
+  it 'registers an offense when attribute validations have same configuration options' do
+    expect_offense(<<~RUBY)
+      validates :email, presence: true, if: :some_condition?
+                ^^^^^^ Group together all email validations.
+      validates :email, format: { with: /@/ }, if: :some_condition?
+                ^^^^^^ Group together all email validations.
+    RUBY
+  end
+
+  it 'registers an offense when attribute validations have different validation options' do
+    expect_offense(<<~RUBY)
+      validates :email, format: { with: /foo/ }
```

Comment:
What would be the expected correction for this case?

```rb
validates :bool, inclusion: { in: [false], if: :foo? }
validates :bool, inclusion: { in: [true], if: :bar? }
```
I can't come up with something that doesn't make this more complicated.

Reply:
```ruby
validates :bool, inclusion: { in: :some_meaningfull_method_name }

def some_meaningfull_method_name
  if foo?
    [false]
  else
    [true]
  end
end
```